BetterMenus/better-menus.md

```diff
@@ -0,0 +1,192 @@
+1. `MenuScreen` It is the object that MAPI needs to display your screen to the user. The final return statement to create a modmenu requires you to return a menuscreen
+2. `Menu` It is the object that satchel created to make making menus easier. This will be referred to as `MenuRef` to avoid confusion. 
+3. `Menu Element` It is the parts that make up the menu (buttons, keybinds etc.)
+
+To Create Menus:
+Make the class that inherits from `Mod` also inherit from `ICusomMenuMod`. You'll be asked to create a function `GetMenuScreen`. In that function do:
+```cs
+// a variable that holds our Satchel.BetterMenu.Menu for us to use in the code.
+private Menu MenuRef;
+
+//function required to be created when inheriting from ICustomMenuMod.
+public MenuScreen GetMenuScreen(MenuScreen modListMenu, ModToggleDelegates? modtoggledelegates) 
+{
+    //Create a new MenuRef if it's not null
+    MenuRef ??= new Menu(
+                "My Mod Name", //the title of the menu screen, it will appear on the top center of the screen 
+                new Element[]
+                {
+                    //add elements here  
+                }
+    );
+    
+    //uses the GetMenuScreen function to return a menuscreen that MAPI can use. 
+    //The "modlistmenu" that is passed into the parameter can be any menuScreen that you want to return to when "Back" button or "esc" key is pressed 
+    return MenuRef.GetMenuScreen(modListMenu);
+}
+```
+
+where you see `//add elements here` you can add elements based on what you need. the 7 elements will be detailed on [Better Menu Elements Page](elements.md)
+1. [Horizontal Option](elements.md#1-horizontal-option)
+2. [Menu Button](elements.md#2-menu-button)
+3. [Text Panel](elements.md#3-text-panel)
+4. [Key Bind](elements.md#4-key-bind)
+5. [Button Bind](elements.md#5-button-bind)
+6. [Custom Slider](elements.md#6-custom-slider)
+7. [Static Panel](elements.md#7-static-panel)
+
+To add lets say a MenuButton that logs when  that button was pressed you would do
+```cs
+new Element[]
+{
+    new MenuButton("My Logging Button", "A menu button", (_) => Log("A button was pressed")),
+}
+```
+
+Below that you can add how many elements you want using `new {Element Name}(),`. If the element count exceeds 5, satchel will create scrollbar for you.
+> Notice how I didn't manually specify an Id. If it is not specified, the Id will be the name of the element
+### 2. Updating properties of elements
+Lets say you wanted to add another option in HorizontalOption or wanted to change what the menu button description says after you press it.
+All you'd have to do is 
+```cs
+//find the element by ID. 
+Element elem = MenuRef.Find("My Logging Button");
+
+//cast the element as a menubutton
+BetterMenus.MenuButton buttonElem = elem as BetterMenus.MenuButton;
+
+//change a property of the element
+buttonElem.Description = "My New MenuButton";
+
+//update menu to reflect changes
+MenuRef.Update();
+```
```

Comment:
i'd always prefer Element.Update because of lower cost on bigger menus, should we show that here?

Reply:
I forgot we can do that 😶